ChangeLog.d/remove-rsa-mode-parameter.txt

```diff
@@ -0,0 +1,15 @@
+Removals
+   * The RSA module no longer supports private-key operations with the public
+     key and vice versa. This change only affects applications which use the
+     wrong mode. In this case the wrong mode is to use mode=MBEDTLS_RSA_PUBLIC
+     with decryption and signing functions and mode=MBEDTLS_RSA_PRIVATE with
+     encryption and verification functions. Addresses issue #4278.
+API changes
+   * Remove mode parameter from RSA functions. All encryption,
+     decryption, sign and verify functions are affected. Also
```

Comment:
I think this would flow better, and would explicitly state what is still supported:

> Remove the mode parameter from RSA operation functions. Signature and decryption functions now always use the private key and verification and encryption use the public key. Verification functions also no longer have RNG parameters.